libs/providers/langchain-xai/README.md

```diff
@@ -31,6 +31,158 @@ const message = new HumanMessage("What color is the sky?");
 const res = await model.invoke([message]);
 ```
 
+## Server Tool Calling (Live Search)
+
+xAI supports server-side tools that are executed by the API rather than requiring client-side execution. The `live_search` tool enables the model to search the web for real-time information.
+
+### Using the built-in live_search tool
+
+```typescript
+import { ChatXAI, tools } from "@langchain/xai";
+
+const model = new ChatXAI({
+  model: "grok-2-1212",
+});
+
+// Create the built-in live_search tool with optional parameters
+const searchTool = tools.xaiLiveSearch({
+  max_search_results: 5,
+  return_citations: true,
```

Comment:
I'ld suggest these options are camelcase and translated within the function to the actual properties.
```suggestion
  maxSearchResults: 5,
  returnCitations: true,
```